packages/gestalt-codemods/generic-codemods/utils.js

```diff
@@ -5,261 +5,278 @@ import {
-    .map((attr) => {
-      const propName = attr?.name?.name;
+const buildReplaceWithModifiedAttributes = ({
+  j,
+  nextProp,
+  nextValue,
+}: {
+  j: JSCodeShift,
+  nextProp?: string,
+  nextValue?: string,
+}): ((nodepath: Collection) => ?JSXNodeType) => {
+  const replaceWithModifiedAttributes = (nodepath: Collection) => {
+    if (!nextProp) return null;
 
-      if (propName !== previousPropName) return attr;
+    let newNode = deepCloneNode({ node: nodepath.get().node });
 
-      const renamedAttr = { ...attr };
+    if (nextProp && isNullOrUndefined(nextValue)) newNode.name.name = nextProp;
 
-      if (nextPropName) renamedAttr.name.name = nextPropName;
+    if (nextProp && !isNullOrUndefined(nextValue))
+      newNode = buildAttributeFromValue({ j, prop: nextProp, value: nextValue });
 
-      return nextPropName !== null && nextPropName ? renamedAttr : undefined;
-    })
-    .filter(Boolean);
+    return newNode;
+  };
 
-type ReplaceJSXAttributesType = {|
-  JSXNode: JSXNodeType,
-  newAttributes: Array<JSXAttributeType>,
-|};
+  return replaceWithModifiedAttributes;
+};
 
 /**
- * replaceJSXAttributes: Saves the changes in the file  if the src object contains the 'modified: true' key-value
+ * throwErrorIfSpreadProps: Throws an error message if component contains spread props which are opaque to codemods
+ * E.g. <Box {...props} /> // error!
  */
-const replaceJSXAttributes = ({ JSXNode, newAttributes }: ReplaceJSXAttributesType): void => {
-  // TODO: implement deep cloning or prevent reassignment
-  const newJSXNode = { ...JSXNode };
+const throwErrorIfSpreadProps = ({
+  fileInfo,
+  j,
+  jSXCollection,
+}: {
+  fileInfo: FileType,
+  j: JSCodeShift,
+  jSXCollection: Collection,
+}): void => {
+  const spreadPropsCollection = jSXCollection.find(j.JSXSpreadAttribute);
 
-  newJSXNode.openingElement.attributes = newAttributes;
+  if (spreadPropsCollection.size() > 0) {
+    throw new Error(
+      `Remove dynamic properties and rerun codemod.\n${spreadPropsCollection
+        .nodes()
+        .map((node) => `Location: ${fileInfo.path} @line: ${node.loc.start.line}`)
+        .join('\n')}`,
+    );
+  }
 };
 
-type SaveSourceType = {| src: Collection |};
-
 /**
- * saveSource: Saves the changes in the file  if the src object contains the 'modified: true' key-value
- */ const saveSource = ({ src }: SaveSourceType): string | null =>
-  src.modified ? src.toSource({ quote: 'single' }) : null;
-
-type ThrowErrorIfSpreadType = {| file: FileType, JSXNode: JSXNodeType |};
-
-/**
- * throwErrorIfSpreadProps: Throws an error message if component contains spread props which are opaque to  codemods
- * E.g. <Box {...props} /> // error!
+ * throwErrorMessage: Throws an error message
  */
-const throwErrorIfSpreadProps = ({ file, JSXNode }: ThrowErrorIfSpreadType): void => {
-  if (
-    JSXNode.openingElement.attributes.some((attribute) => attribute.type === 'JSXSpreadAttribute')
-  ) {
+const throwErrorMessage = ({
```

Comment:
nit: this name doesn't quite cover what the function does. It makes it seem like this will always throw, when it's actually only going to throw conditionally